### DISCUSSION TITLE
How can I see the generated source files?

### DISCUSSION THREAD

**Discussion by Shayan-To** [dotnet/roslyn#48811] [Q&A] [answered]
I have written a source generator, and it is working well. But I want to check the generated source code to see whether it actually is what I want it to be (and to test some edge cases). I couldn't find any way to see what my generator is producing.

What am I missing? Is there a switch I need to turn on?

**Comment by PathogenDavid** [comment, marked as answer]
Add `<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>` to your project properties and build.

The files will appear in your intermediate directory in a folder called `generated`. (IE: `path/to/project/obj/Debug/net5.0/generated/`)

(Note that this is only really intended for debugging, as noted in [the discussion when this feature was added](https://github.com/dotnet/roslyn/pull/47047).)

**Comment by zbesevic** [comment]
How can I add these files to the repository? Is the path fixed?

**Reply by PathogenDavid** [reply]
You really probably shouldn't be committing them to source control. The feature is only intended for debugging, and is explicitly not intended to provide any sort of caching mechanism. ([See the discussion when this change was added](https://github.com/dotnet/roslyn/pull/47047)) If your source generator takes a long time to run, you might consider whether it's the right tool for the job, caching something like source generator output is pretty easy to get incorrect and cause weird build failures.

That being said, if that didn't scare you off you can change `CompilerGeneratedFilesOutputPath` to set where the files get written.

**Reply by zbesevic** [reply]
Tnx works. Is there a way to define the folder names where the generated files will be saved (directory structure)?

**Reply by PathogenDavid** [reply]
I'm not entirely sure what you mean by that.

Like you want to put the files in a different folder for debug and release configurations? If yes, use something like `<CompilerGeneratedFilesOutputPath>generated/$(Configuration)/<CompilerGeneratedFilesOutputPath>`.

If you meant something else, can you give an example?
